src/Controller/Router.php

```diff
@@ -456,6 +464,10 @@ protected function setStoreConfigs(ActionInterface $action)
             $storeConfig[$configKey] = $configValue;
         }
 
+        if (isset($storeConfig[self::TAX_DISPLAY_TYPE_CONFIG_KEY])) {
```

Comment:
Can we please move this logic to foreach block, so it we kind of won't be needed to override it after it. Otherwise a bit of magic from first look.